File: java/test/src/main/java/test/Freeze/fileLock/ClientFail.java
// **********************************************************************
//
// Copyright (c) 2003-present ZeroC, Inc. All rights reserved.
//
// **********************************************************************

package test.Freeze.fileLock;

public class ClientFail extends test.TestHelper
{
    @Override
    public void run(String[] args)
    {
        Ice.StringSeqHolder argsH = new Ice.StringSeqHolder(args);
        Ice.Properties properties = createTestProperties(argsH);
        try(Ice.Communicator communicator = initialize(properties))
        {
            String envName = "db";
            if(argsH.value.length > 0)
            {
                envName = argsH.value[0];
                envName += "/";
                envName += "db";
            }

            System.out.println("create createConnection");
            Freeze.Util.createConnection(communicator, envName);
            test(false);
        }
        catch(Freeze.FileLockException ex)
        {
            System.out.println("File lock not acquired.");
        }
    }
}

File: cpp/test/Freeze/fileLock/ClientFail.cpp
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

#include <IceUtil/IceUtil.h>
#include <Freeze/Freeze.h>
#include <TestHelper.h>

using namespace std;
using namespace Ice;
using namespace Freeze;

class ClientFail : public Test::TestHelper
{
public:

    void run(int, char**);
};

void
ClientFail::run(int argc, char** argv)
{
    string envName = "db";

    Ice::CommunicatorHolder ich = initialize(argc, argv);
    if(argc != 1)
    {
        envName = argv[1];
        envName += "/db";
    }

    {
        try
        {
            Freeze::createConnection(communicator(), envName);
            test(false);
        }
        catch(const IceUtil::FileLockException&)
        {
            cout << "File lock not acquired." << endl;
        }
    }
}

DEFINE_TEST(ClientFail)
